Simplified/Reader2/LCP/LCPPassphraseAuthenticationService.swift

```diff
@@ -0,0 +1,46 @@
+//
+//  LCPPassphraseAuthenticationService.swift
+//  Simplified
+//
+//  Created by Ernest Fan on 2021-02-08.
+//  Copyright © 2021 NYPL Labs. All rights reserved.
+//
+
+#if LCP
+
+import Foundation
+import ReadiumLCP
+
+class LCPPassphraseAuthenticationService: LCPAuthenticating {
+  func retrievePassphrase(for license: LCPAuthenticatedLicense, reason: LCPAuthenticationReason, allowUserInteraction: Bool, sender: Any?, completion: @escaping (String?) -> Void) {
+    guard let hintLink = license.hintLink,
+      let hintURL = URL(string: hintLink.href) else {
+      Log.error(#file, "LCP Authenticated License does not contain valid hint link")
+      completion(nil)
+      return
+    }
+    
+    NYPLNetworkExecutor.shared.GET(hintURL) { (result) in
+      switch result {
+      case .success(let data, _):
+        if let json = try? JSONSerialization.jsonObject(with: data, options: []) as! [String:Any],
+          let passphrase = json["passphrase"] as? String {
```

Comment:
do we have docs for the format of the response? For example that you're looking for that specific `passphrase` key.  If it's documented somewhere it would be good to link it somewhere in this class.

Reply:
It's not on Readium's requirement, but I will check with the backend developer.
https://readium.org/lcp-specs/releases/lcp/latest#41-introduction

Reply:
sounds good -- that link is a good thing to add though.